zilliqa/src/contracts/deposit_v1.sol

```diff
@@ -0,0 +1,293 @@
+    // The index of the value in the `stakers` array plus 1.
+    // Index 0 is used to mean a value is not present.
+    uint256 index;
+    // Invariant: `balance >= minimumStake`
+    uint256 balance;
+}
+
+struct Committee {
+    // Invariant: Equal to the sum of `balances` in `stakers`.
+    uint256 totalStake;
+    bytes[] stakerKeys;
+    mapping(bytes => CommitteeStakerEntry) stakers;
+}
+
+struct Staker {
+    // The address used for authenticating requests from this staker to the deposit contract.
+    // Invariant: `controlAddress != address(0)`.
+    address controlAddress;
+    // The address which rewards for this staker will be sent to.
+    address rewardAddress;
+    // libp2p peer ID, corresponding to the staker's `blsPubKey`
+    bytes peerId;
+    // Invariants: Items are always sorted by `startedAt`. No two items have the same value of `startedAt`.
+    Deque.Withdrawals withdrawals;
+}
+
+// Parameters passed to the deposit contract constructor, for each staker who should be in the initial committee.
+struct InitialStaker {
+    bytes blsPubKey;
+    bytes peerId;
+    address rewardAddress;
+    address controlAddress;
+    uint256 amount;
+}
+
+contract DepositInit is UUPSUpgradeable, Ownable2StepUpgradeable {
+    /// @custom:storage-location erc7201:zilliqa.storage.DepositStorage
+    struct DepositStorage {
+        // The committee in the current epoch and the 2 epochs following it. The value for the current epoch
+        // is stored at index (currentEpoch() % 3).
+        Committee[3] _committee;
+        // All stakers. Keys into this map are stored by the `Committee`.
+        mapping(bytes => Staker) _stakersMap;
+        // Mapping from `controlAddress` to `blsPubKey` for each staker.
+        mapping(address => bytes) _stakerKeys;
+        // The latest epoch for which the committee was calculated. It is implied that no changes have (yet) occurred in
+        // future epochs, either because those epochs haven't happened yet or because they have happened, but no deposits
+        // or withdrawals were made.
+        uint64 latestComputedEpoch;
+        uint256 minimumStake;
+        uint256 maximumStakers;
+        uint64 blocksPerEpoch;
+    }
+
+    // keccak256(abi.encode(uint256(keccak256("zilliqa.storage.DepositStorage")) - 1)) & ~bytes32(uint256(0xff))
+    bytes32 private constant DepositStorageLocation =
+        0x958a6cf6390bd7165e3519675caa670ab90f0161508a9ee714d3db7edc507400;
+
+    function _getDepositStorage()
+        private
+        pure
+        returns (DepositStorage storage $)
+    {
+        assembly {
+            $.slot := DepositStorageLocation
+        }
+    }
+
+    function version() public view returns (uint64) {
+        return _getInitializedVersion();
+    }
+
+    function __Deposit_init(address initialOwner) internal onlyInitializing {
+        __Ownable2Step_init_unchained();
+        __Ownable_init_unchained(initialOwner);
+        __UUPSUpgradeable_init_unchained();
```

Comment:
This pattern with the `__ContractName_init()` is only used in base contracts i.e. if other contracts inherited from the `Deposit` contract but this is not the case. So we can get rid of this function and call `__UUPSUpgradeable_init_unchained();`  in the `initialize()` function. As I mentioned below, I'd remove `Ownable2StepUpgradeable` from the base contracts so that we don't need the other two calls.